### PR TITLE
Clarify that "Depends on" should be in summary

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: arcanist-user.rst [hunk 1]
```diff
@@ -243,10 +243,25 @@ shows the current stack of revisions:
    :align: center
    :alt: Screenshot of a revision stack
 
-You can also add ``Depends on D<revision ID>`` to the child's commit
-message, replacing ``<revision ID>`` with the ID of the parent
+You can also add ``Depends on D<revision ID>`` to the child's revision
+summary, replacing ``<revision ID>`` with the ID of the parent
```
Comment (markrcote) [inline arcanist-user.rst]: My only hesitation here is that the revision summary is created from the commit message. I fear that not mentioning the commit message will cause confusion, e.g. "how do I put stuff into the revision summary??".

File: arcanist-user.rst [hunk 2]
```diff
@@ -243,10 +243,25 @@ shows the current stack of revisions:
...
 revision. (This needs to be its own paragraph, separated by a blank line.)
-The relationship will be created when ``arc diff`` is run.
+The relationship will be created when ``arc diff`` is run.::
+
+    ... commit comment ...
```
Comment (markrcote) [inline arcanist-user.rst]: This is actually the commit summary, i.e. the first line of the commit message. The rest of the comment will go into the Summary field below.

File: arcanist-user.rst (hunk 1 above)
Comment (djmitche) [reply arcanist-user.rst]: I share that confusion.  Can we define "revision summary" "commit message 'and "commit summary"?

Comment (markrcote) [reply arcanist-user.rst]: In this doc, you mean?

Comment (djmitche) [reply arcanist-user.rst]: At least in the PR comments -- I don't understand the distinction you're making.

Comment (markrcote) [reply arcanist-user.rst]: Ah okay:

- **commit message**: the entire message in the commit, which appears in the full {hg, git} logs
- **commit summary**: the first line of a commit message (I don't think this is an official definition, but we often refer to it. Also I am just now realizing this is confusing given the next definition.)
- **revision summary**: the field in a Phabricator revision within the "Details" panel that is labelled "Summary". This is also editable as part of the ``arc diff`` submission process.

Also,
- **revision title**: the top, bold line in a revision, which you would also see in a list of revisions along with the revision ID

When a revision is created via Arcanist, the editor that pops open defaults to setting the revision title from the commit summary and the revision summary from the rest of the commit message. Editing those fields at this point will also update the commit message (if history is set as mutable, which is a Phabricator setting we generally apply to our repos). 

So, long story short, it doesn't _really_ matter if you put the ``Depends on ...`` into the commit message or the revision summary, as it will be mirrored to the other when the revision is submitted via ``arc diff``.

Comment (djmitche) [reply arcanist-user.rst]: OK, so it sounds like no change is needed here.  Thanks!
